### PR TITLE
Integrate dispensary; remove filehasher (close #475)

### LINKED ISSUE #475: Integrate dispensary
This is just a last piece of the puzzle but we should remove the hash checking code in the current linter and switch to dispensary's `match()` API.

### PR DESCRIPTION
**Pull request by tofumatt**
Puts #66 to rest, yay!

As mentioned in #475, we should wait until https://github.com/mozilla/dispensary/issues/43 is closed to merge.

It will actually fail until https://github.com/mozilla/dispensary/issues/42 is merged and I bump the version of dispensary we're using.

### CODE REVIEW

**Comment by tofumatt** [comment]
r?

**Comment by muffinresearch** [comment]
travis-says-no

**Comment by tofumatt** [comment]
Oh damn, how weird. I’ll check it out.

**Comment by tofumatt** [comment]
This should _finally_ pass, so r? :grinning:

File: tests/test.linter.js [hunk 1]
```diff
@@ -549,7 +549,9 @@ describe('Linter.getAddonMetadata()', function() {
       });
   });
 
-  it('should recognise it as a Jetpack add-on', () => {
+  it('should recognise it as a Jetpack add-on', function() {
+    this.timeout(20000);
```
Comment (muffinresearch) [inline tests/test.linter.js]: Are these epic timeouts needed?

**Comment by muffinresearch** [comment]
r+wc

File: tests/test.linter.js (hunk 1 above)
Comment (tofumatt) [inline tests/test.linter.js]: I’ll check, but I know running the dispensary on add ons with lots of JS files slows things down, but hopefully not that much.

**Pull request merged**